packages/samples/headless-commerce-ssr-remix/lib/commerce-engine.server.ts

```diff
@@ -0,0 +1,101 @@
+import {coveo_accessToken} from '@/app/cookies.server';
+import externalCartService from '@/external-services/external-cart-service';
+import externalContextService from '@/external-services/external-context-service';
+import {
+  toCoveoCartItems,
+  toCoveoCurrency,
+} from '@/utils/external-api-conversions';
+import {
+  NavigatorContext,
+  SolutionType,
+} from '@coveo/headless-react/ssr-commerce';
+import {isExpired} from '../utils/access-token-utils';
+import {
+  _listingEngineDefinition,
+  _recommendationEngineDefinition,
+  _searchEngineDefinition,
+  _standaloneEngineDefinition,
+  ListingEngineDefinition,
+  RecommendationEngineDefinition,
+  SearchEngineDefinition,
+  StandaloneEngineDefinition,
+} from './commerce-engine';
+import {fetchToken} from './fetch-token';
+
+type MappedEngineDefinition<T extends SolutionType> =
```

Comment:
Can this be part of the headless-ssr package instead?

Reply:
Ultimately, yes!

However, the pattern I'm suggesting here might very well evolve as we discover new things about Headless commerce SSR implementations and keep improving our samples... So maybe we should not include this type in the public API just yet?